controllers/selfnoderemediation_controller.go

```diff
@@ -154,13 +154,20 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, err
 	}
 
-	snrOrg := snr.DeepCopy()
 	defer func() {
-		if patchErr := r.patchSnrStatus(ctx, snr, snrOrg); patchErr != nil {
-			if returnErr == nil {
-				returnErr = patchErr
+		if updateErr := r.updateSnrStatus(ctx, snr); updateErr != nil {
+			if apiErrors.IsConflict(updateErr) {
+				if returnErr == nil {
+					if !returnResult.Requeue {
```

Comment:
Hmm I'm not sure, the change isn't logically equal to the original code.
Those are the differences I've found:
- In the changed code we will aggregate a conflict error to an existing error 
- In the changed code (assuming conflict but no original error)  we are overriding the original requeue value (in case there was one).

Reply:
Already done, this code is outdated :slightly_smiling_face:

Reply:
I think the latter is important ^^
Failed status update should always requeue quickly

/hold